docs/design/interpreter/ir-byte-code.md

```diff
@@ -0,0 +1,399 @@
+# The IR byte code
+## Introduction
+ * There are 677 byte codes vs 263 IL opcodes 
+ * Some of them are handling intrinsics (includes also stuff like String.Length, System.Diagnostics.Debugger.Break)
+ * Each opcode is directly followed by arguments that have variable size depending on the opcode.
+ * Both the transformation and interpretation use Mono type / method  / vtable etc data structures
+ * The arguments to the opcode are immediate constants (16, 32 or 64 bit), indexes into local variables table or indexes into a per-method special data items array.
+## Limitations
+ * The local variable offsets in most of the IR codes are 16 bit unsigned integers, so a method can have max 65536 kB of local variables
```

Comment:
i think this is a 'max stack frame size of 65kb', not just local variables since stuff like stackalloc eats the space too